src/plugins/announce.lua

```diff
@@ -0,0 +1,53 @@
+-- URL announcements - follows http->https redirects but not www
+
+local https = require 'ssl.https'
+local ltn12 = require 'ltn12'
+
+local plugin = {}
+
+plugin.help = 'Usage: announce <url>'
+
+plugin.main = function(args)
+    local _, _, url = args.message:find('(.+)')
+
+    if args.modules.config.debug then
+        print(url)
+    end
+
+    if not url then
+        args.modules.irc.privmsg(args.target, ('%s: Please give me a url.'):format(args.sender))
+        return
+    end
+
+    if url:find('[ <]?https?://[^> ]+.*') then
+        url = url:match('[ <]?https?://[^> ]+.*')
+    elseif url:find(' www%..*') then
+        url = url:match(' www%..*')
+        url = 'http://' .. url
+    else
+        url = 'http://www.' .. url
```

Comment:
Given the removal of the `elseif` block and the extension of the `if` block, I still think this `else` case makes sense, but I disagree with prepending `www`. Rather, I would only prepend `http://`. This is dramatically less likely to cause issues.

Reply:
During testing not having www caused more problems than having it. The majority of major websites redirect the base domain to www.domain (and the code cannot follow the redirect) for SEO purposes and cookie handling. (cookies set for www don't affect other subdomains like mail.domain but they do affect them if set on the base domain)

I'm very happy to change this, but the above was my reasoning for doing it. Basically it was a choice to have lots of unfollowed 301 redirects get returned or have a `www` default that causes an error message on domains that don't use www or redirect www to the base domain.

Reply:
Personally, I line up on the side of “this is an upstream issue” (and, in this case, actually has a merged fix iirc, we are just waiting on a new release). Not having it means that unadorned URLs that don't follow the `www` convention are still possible.